### PR TITLE
Add copyright notice to all source files

### PR DESCRIPTION
**Pull request by dberzano**

### CODE REVIEW

**Comment by dberzano** [comment]
Please do not merge yet. Work is in progress.

@Barthelemy @ktf @sawenzel

**Comment by dberzano** [comment]
Goes together with alisw/alidist#754

**Comment by matthiasrichter** [comment]
We still have the problem that https://alice-o2.web.cern.ch/ is a web page which requires login, @Barthelemy can that be fixed? We would need a different link in the copyright notice otherwise. Also I can not find the licensing information on this web page.

File: DataFormats/Headers/include/Headers/DataHeader.h [hunk 1]
```diff
@@ -1,3 +1,13 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See https://alice-o2.web.cern.ch/ for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
```
Comment (matthiasrichter) [inline DataFormats/Headers/include/Headers/DataHeader.h]: now we have conflicting copyright notices in many files

**Comment by dberzano** [comment]
Hi @matthiasrichter, the licensing information and its inclusion in all source files was approved in a O2 plenary following a work from CWG2 first, CWG11 then. Minor modifications to the original text were added following a suggestion from the CERN legal office and KT, taking into account contributors external to CERN.

The PR along with alisw/alidist#754 is a start, and forces all new contributions to have the correct licensing information. I'd say we can clean up old licenses as we go. On a separate note, we have decided to proceed during our last WP3 meeting where only two persons were present.

Let's keep track of further issues concerning copyright in [OCCF-34](https://alice.its.cern.ch/jira/browse/OCCF-34).

**Comment by sawenzel** [comment]
@dberzano: It is still preferable if a merge request gets merged by someone else than the author. This forces at least a minimal crosscheck.

**Comment by matthiasrichter** [comment]
Hi Dario, all good, and thanks for your effort. +1 also for the new CI check.

I disagree on the fact to have duplicated and probably conflicting license information in the files. Also one should make sure that the text added as new notice is consistent with what it refers to before merging a PR like this. There is neither a file `COPYING` in the repository nor does the cited web page lead to any information, ignoring now the fact that the page is not publicly available while the code is. Actually I couldn't care less but after the long discussion with many valid points raised, good suggestions, and a lot of time spent on this, I'm not fully satisfied with the outcome.

**Comment by dberzano** [comment]
Hi all,

1. the "having a copyright notice at the beginning of each file" problem was factored out from the rest - let's now concentrate on fixing what's wrong and addressing other copyright problems in [OCCF-34](https://alice.its.cern.ch/jira/browse/OCCF-34).

2. true, we should have one person's PR approved by other people, in principle, however what's here does not break O2, it's formally correct, and as said already - was decided by the Plenary

3. sorry for merging this but there was a gigantic pressure from the steering board to get this initial part done, and given that we have been discussing this for more than one year, and a decision was eventually taken, the SB was right in asking this to be done ASAP

**Comment by Barthelemy** [comment]
@matthiasrichter The page is accessible if you use "http" instead of "https". https://alice.its.cern.ch/jira/browse/OCCF-35

**Pull request merged**